ts/Grid/Core/Pagination/Pagination.ts

```diff
@@ -305,22 +911,49 @@ class Pagination {
      * Reflow the pagination container.
      */
     public reflow(): void {
-        if (!this.cell) {
+        const position = this.options.position;
+
+        if (!this.paginationContainer) {
             return;
         }
 
-        this.cell.style.width = this.grid.tableElement?.offsetWidth + 'px';
+        if (position === 'footer') {
+            // Set the width to match the table width
+            this.paginationContainer.style.width =
+                this.grid.tableElement?.offsetWidth + 'px';
+            return;
+        }
     }
 
     /**
-     * The pagination will be removed from the container.
+     * Destroy the pagination instance.
      */
     public destroy(): void {
-        this.row?.remove();
+        const position = this.options.position;
+
+        if (position === 'footer') {
+            // For footer position, remove the entire tfoot element.
+            this.paginationContainer?.parentElement?.parentElement?.remove();
+        } else if (typeof position === 'string' && position.startsWith('#')) {
+            // Don't remove the custom container itself
+            this.contentWrapper?.remove();
+        } else {
+            // For top/bottom positions, remove the entire pagination container
+            this.paginationContainer?.remove();
```

Comment:
Are you sure this works?
It does not look like `this.paginationContainer` is set when using `top/bottom`

Reply:
in the `render()`